front/lib/resources/retrieval_document_resource.ts

```diff
@@ -0,0 +1,173 @@
+// Attributes are marked as read-only to reflect the stateless nature of our Resource.
+// This design will be moved up to BaseResource once we transition away from Sequelize.
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+
+import type {
+  ModelId,
+  Result,
+  RetrievalDocumentChunkType,
+  RetrievalDocumentType,
+} from "@dust-tt/types";
+import type { Attributes, CreationAttributes, ModelStatic } from "sequelize";
+
+import config from "@app/lib/api/config";
+import type { Authenticator } from "@app/lib/auth";
+import {
+  RetrievalDocument,
+  RetrievalDocumentChunk,
+} from "@app/lib/models/assistant/actions/retrieval";
+import { BaseResource } from "@app/lib/resources/base_resource";
+import { frontSequelize } from "@app/lib/resources/storage";
+import type { ReadonlyAttributesType } from "@app/lib/resources/storage/types";
+
+export type RetrievalDocumentBlob = CreationAttributes<RetrievalDocument>;
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-declaration-merging
+export interface RetrievalDocumentResource
+  extends ReadonlyAttributesType<RetrievalDocument> {}
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-declaration-merging
+export class RetrievalDocumentResource extends BaseResource<RetrievalDocument> {
+  static model: ModelStatic<RetrievalDocument> = RetrievalDocument;
+
+  constructor(
+    model: ModelStatic<RetrievalDocument>,
+    blob: Attributes<RetrievalDocument>,
+    readonly chunks: RetrievalDocumentChunkType[]
+  ) {
+    super(RetrievalDocument, blob);
+  }
+
+  static async makeNew(
+    auth: Authenticator,
+    blob: RetrievalDocumentBlob,
+    chunks: RetrievalDocumentChunkType[]
+    // TODO(GROUPS_INFRA) Add supports for dataSourceViewId.
+  ) {
+    const [doc] = await this.makeNewBatch(auth, [{ blob, chunks }]);
+
+    return doc;
+  }
+
+  static async makeNewBatch(
+    auth: Authenticator,
+    blobs: {
+      blob: RetrievalDocumentBlob;
+      chunks: RetrievalDocumentChunkType[];
+      // TODO(GROUPS_INFRA) Add supports for dataSourceViewId.
+    }[]
+  ) {
+    // TODO(GROUPS_INFRA) Use auth.workspaceId.
+    const results = await frontSequelize.transaction(async (transaction) => {
+      const createdDocuments = await Promise.all(
+        blobs.map(async ({ blob, chunks }) => {
+          const doc = await RetrievalDocument.create(blob, { transaction });
+
+          await Promise.all(
```

Comment:
Why do we need to promise.all ? old version wasn't doing that, we should keep the existing logic IMO (unless we think there is something to fix)